### PR TITLE
Add python package from protos

### PR DESCRIPTION
**Pull request by ngruelaneo**
fix https://github.com/aneoconsulting/ArmoniK/issues/468

### CODE REVIEW

File: proto2python.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+armonik_common_files=("objects.proto" "task_status.proto" "session_status.proto" \
+                      "result_status.proto" "agent_common.proto" "sessions_common.proto"  \
+                      "submitter_common.proto"  "tasks_common.proto"  "worker_common.proto")
+
+python -m pip install --upgrade pip
+python -m venv $HOME/grpc2
+python -m pip install build
+python -m pip install grpcio grpcio-tools fix-protobuf-imports
+
+for proto in ${armonik_server_files[@]}; do
```
Comment (dbrasseur-aneo) [inline proto2python.sh:30]: You should be able to generate with all files at once for each category

File: proto2python.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+# Bash script to create the python packages from the grpc proto for Armonik.Api
+# We are using the same structure than the C# package
+
+export PATH=$HOME/.local/bin:$PATH
+
+export PROTO_PATH="Protos/V1"
+export ARMONIK_PYTHON_SRC="Api/python/src" 
+export ARMONIK_SERVER=$ARMONIK_PYTHON_SRC"/armonik/server"
```
Comment (dbrasseur-aneo) [inline proto2python.sh:10]: Naming is ambiguous
Maybe WORKER would be better. This is for the worker side of ArmoniK but has both grpc client and server components

**Review by dbrasseur-aneo — CHANGES_REQUESTED** [review]

File: proto2python.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+for proto in ${armonik_client_files[@]}; do
+    python -m grpc_tools.protoc -I $PROTO_PATH --proto_path=$PROTO_PATH \
+            --python_out=$ARMONIK_CLIENT --grpc_python_out=$ARMONIK_CLIENT \
+            $PROTO_PATH/$proto
+done
+
+for proto in ${armonik_common_files[@]}; do
+    python -m grpc_tools.protoc -I $PROTO_PATH --proto_path=$PROTO_PATH \
+            --python_out=$ARMONIK_COMMON --grpc_python_out=$ARMONIK_COMMON \
```
[the remark this answers is not in the document]
Comment (ngruelaneo) [reply proto2python.sh:44]: We need it, at least for now. It used by the script to cancel the tasks.

File: proto2python.sh (hunk 2 above)
Comment (ngruelaneo) [reply proto2python.sh:10]: I am using the same naming thant the extension CSharp (in theory)

File: proto2python.sh [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+# sudo apt install python3-venv
+
+armonik_server_files=("agent_service.proto" "worker_service.proto")
+armonik_client_files=("submitter_service.proto" "tasks_service.proto" "sessions_service.proto")
+armonik_common_files=("objects.proto" "task_status.proto" "session_status.proto" \
+                      "result_status.proto" "agent_common.proto" "sessions_common.proto"  \
+                      "submitter_common.proto"  "tasks_common.proto"  "worker_common.proto")
+
+python -m pip install --upgrade pip
+python -m venv $HOME/grpc2
```
[the remark this answers is not in the document]
Comment (ngruelaneo) [reply proto2python.sh:26]: There are nothing happening and it should be done on the CI/CD so it does not matter too much. It is a one of.

File: proto2python.sh (hunk 1 above)
Comment (ngruelaneo) [reply proto2python.sh:32]: done

File: proto2python.sh [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            $PROTO_PATH/$proto
+done
+
+for proto in ${armonik_common_files[@]}; do
+    python -m grpc_tools.protoc -I $PROTO_PATH --proto_path=$PROTO_PATH \
+            --python_out=$ARMONIK_COMMON --grpc_python_out=$ARMONIK_COMMON \
+            $PROTO_PATH/$proto
+done
+
+for proto in ${armonik_server_files[@]}; do
```
[the remark this answers is not in the document]
Comment (ngruelaneo) [reply proto2python.sh]: erased

File: proto2python.sh (hunk 2 above)
Comment (dbrasseur-aneo) [reply proto2python.sh:10]: In extension.csharp, it's worker

File: proto2python.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (ngruelaneo) [reply proto2python.sh]: Done

**Review by aneojgurhem — APPROVED** [review]

**Review by aneojgurhem — DISMISSED** [review]

**Review by aneojgurhem — APPROVED** [review]

**Review by dbrasseur-aneo — APPROVED** [review]

**Pull request merged**